build.rs

```diff
@@ -46,21 +49,59 @@ fn copy_shared_static_files() {
     }
 }
 
-fn cargo_build_in_directory(directory: &str) -> IoResult<ExitStatus> {
+// Looks for an environmental variable, defaulting to the empty string.
+fn get_env(key: &str) -> String {
+    env::var(key).unwrap_or("".to_owned())
+}
+
+// Returns the appropriate --target=XXX command line argument, or None
+// if we are not cross compiling.
+fn get_target() -> Option<String> {
+    let target = get_env("TARGET");
+    let host = get_env("HOST");
+    if host != target {
+        return Some(format!("--target={}", target));
+    }
+
+    None
+}
+
+fn cargo_build_in_directory(directory: &str) {
     let current_dir = env::current_dir().unwrap();
     let mut run_in_dir = PathBuf::from(&current_dir);
     run_in_dir.push(directory);
 
-    Command::new("cargo")
-            .arg("build")
-            .current_dir(run_in_dir)
-            .spawn()
-            .unwrap()
-            .wait()
+    let mut exec = Command::new("cargo");
+    let mut command = exec.arg("build")
+                          .arg("--verbose"); // Ideally should reflect the --verbose flag of the main
+                                             // cargo command.
+
+    let target = get_target();
+    if target.is_some() {
+        let tmp = command;
+        command = tmp.arg(target.unwrap());
+    }
+
+    // $PROFILE is either `debug` (default value) or `release`.
+    if get_env("PROFILE") == "release" {
+        let tmp = command;
+        command = tmp.arg("--release");
+    }
+
+    let output = command.current_dir(run_in_dir)
+                        .output()
+                        .unwrap();
+
+    if !output.status.success() {
```

Comment:
Why display the output only if the command was successful?
It might be useful to see the output of this command regardless of success state, by default, stdin, stdout and stderr are inherited from the parent with `spawn` on a `Command`.